middleman/media_responses.py

```diff
@@ -0,0 +1,176 @@
+                    else:
+                        management_room_text = f"Failed to send {media_name[self.media_type]} back to sender:" \
+                            f"{response.message if isinstance(response, RoomSendError) else response}"
+                    logger.warning(management_room_text)
+                # Confirm in management room
+                if self.config.confirm_reaction:
+                    await send_reaction(
+                        self.client,
+                        self.room.room_id,
+                        self.event.event_id,
+                        management_room_text
+                    )
+                else:
+                    await send_text_to_room(
+                        self.client,
+                        self.room.room_id,
+                        management_room_text,
+                        True,
+                    )
+            else:
+                logger.debug(
+                    f"Skipping {media_name[self.media_type]} {self.event.event_id} which is not a reply to one of our relay messages",
+                )
+        else:
+            logger.debug(f"Skipping {self.event.event_id} reply {media_name[self.media_type]}")
+
+    def is_mention_only_room(self, identifiers: List[str], is_named: bool) -> bool:
+        """
+        Check if this room is only if mentioned.
+        """
+        if self.config.mention_only_always_for_named and is_named:
+            return True
+        for identifier in identifiers:
+            if identifier in self.config.mention_only_rooms:
+                return True
+        return False
+
+    async def process(self):
+        """
+        Process media.
+        - if management room, identify replies and forward back to original messages.
+        - anything else, relay to management room.
+        """
+        if self.room.room_id == self.config.management_room_id:
+            await self.handle_management_room_media()
+        else:
+            await self.relay_to_management_room()
+
+    async def relay_to_management_room(self):
+        """Relay to the management room."""
+        # First check if we want to relay this
+        if self.is_mention_only_room([self.room.canonical_alias, self.room.room_id], self.room.is_named):
+            # skip media in mention only rooms for now
+            logger.debug(f"Skipping {media_name[self.media_type]} %s in room %s as it's set to only relay on mention and we were "
+                         "not mentioned.", self.event.event_id, self.room.room_id)
+            return
+
+        if self.config.anonymise_senders:
+            text = f"anonymous sent {media_name[self.media_type]}:"
+        else:
+            text = f"{self.event.sender} in {self.room.display_name} (`{self.room.room_id}`) sent {media_name[self.media_type]} " \
+                   f"{self.body}:"
+        response = await send_text_to_room(self.client, self.config.management_room, text, notice=True)
```

Comment:
The "text and then reply with image to it" looks a bit tad weird, but I guess this is somewhat unavoidable, given Matrix doesn't support combining text and images :(

![Selection_210](https://user-images.githubusercontent.com/1174866/190832929-cd9c66bb-38df-4157-a5a7-7d129ba50844.png)